File: lab7/Tree.java

public class Tree extends Plant
{
	private Leaves leaves_type;
	private boolean give_fruits;
	private static int count = 0;
	
	public Leaves get_leaves_type()
	{
		return leaves_type;
	}
	public void set_leaves_type(Leaves leaves_type)
	{
		this.leaves_type = leaves_type;
	}
	
	public boolean get_give_fruits()
	{
		return give_fruits;
	}
	public void set_give_fruits(boolean give_fruits)
	{
		this.give_fruits = give_fruits;
	}
	
	public static int get_count()
	{
		return count;
	}
	
	public Tree()
	{
		latin_name = "Quercus petraea";
		lifespan = 800;
		leaves_type = Leaves.DECIDUOUS;
		give_fruits = false;
		count++;
	}
	protected void finalize()
	{
		count--;
	}
}

File: lab7/Tree.cpp
#include "Tree.h"

int Tree::count = 0;
int Tree::get_count()
{
	return count;
}

leaves Tree::get_leaves_type()
{
	return leaves_type;
}
void Tree::set_leaves_type(leaves leaves_type)
{
	this->leaves_type = leaves_type;
}

bool Tree::get_give_fruits()
{
	return give_fruits;
}
void Tree::set_give_fruits(bool give_fruits)
{
	this->give_fruits = give_fruits;
}

Tree::Tree()
{
	latin_name = "Quercus petraea";
	lifespan = 800;
	leaves_type = DECIDUOUS;
	give_fruits = false;
	count++;
}
Tree::~Tree()
{
	count--;
}